cwltool/job.py

```diff
@@ -278,8 +278,12 @@ def _execute(
                 ]
             ),
             " < %s" % self.stdin if self.stdin else "",
-            " > %s" % os.path.join(self.outdir, self.stdout) if self.stdout else "",
-            " 2> %s" % os.path.join(self.outdir, self.stderr) if self.stderr else "",
+            " > %s" % os.path.join(self.base_path_logs, self.stdout)
+            if self.stdout
+            else "",
+            " 2> %s" % os.path.join(self.base_path_logs, self.stderr)
+            if self.stderr
+            else "",
```

Comment:
Here you could try to capture stdout/stderr logs `if runtimeContext.log_dir != ""`

Reply:
I could.  Not sure what is best.  I could capture logs if someone specifes log_dir but if stderr and stdout are undefined, do you think it could be an issue?  

I can always generate a unique id for stdout and stderr but not sure if that could break tools that specifically don't want to capture stdout/stderr.

I went with an opt in approach right now.